File: VECTOAux/VectoAuxiliaries/Hvac/ISSMTOOL.vb
Imports TUGraz.VectoCommon.Utils
Imports VectoAuxiliaries.Hvac

Public Interface ISSMTOOL
	Property GenInputs As ISSMGenInputs
	Property TechList As ISSMTechList
	Property Calculate As ISSMCalculate
	Property SSMDisabled As Boolean
	Property HVACConstants As IHVACConstants

	ReadOnly Property ElectricalWBase As Double	'Watt
	ReadOnly Property MechanicalWBase As Double	' Watt
	ReadOnly Property FuelPerHBase As Double ' LiterPerHour

	ReadOnly Property ElectricalWAdjusted As Double	' Watt
	ReadOnly Property MechanicalWBaseAdjusted As Double	' Watt
	ReadOnly Property FuelPerHBaseAdjusted As Double ' LiterPerHour

	Sub Clone(from As ISSMTOOL)

	Function Load(filePath As String) As Boolean

	Function Save(filePath As String) As Boolean

	Function IsEqualTo(source As ISSMTOOL) As Boolean

	''' <summary>
	''' This alters the waste heat and returns an adjusted fueling value
	''' </summary>
	''' <param name="AverageUseableEngineWasteHeatKW"></param>
	''' <returns></returns>
	''' <remarks></remarks>
	Function FuelPerHBaseAsjusted(AverageUseableEngineWasteHeatKW As Double) As Double

	Event Message(ByRef sender As Object, message As String, messageType As AdvancedAuxiliaryMessageType)
End Interface

File: VECTOAux/BusAuxiliaries/Interfaces/DownstreamModules/HVAC/ISSMTOOL.cs
namespace TUGraz.VectoCore.BusAuxiliaries.Interfaces.DownstreamModules.HVAC {
	public interface ISSMTOOL
	{
		ISSMGenInputs GenInputs { get; set; }
		ISSMTechList TechList { get; set; }
		ISSMCalculate Calculate { get; set; }
		bool SSMDisabled { get; set; }
		IHVACConstants HVACConstants { get; set; }

		double ElectricalWBase { get; } // Watt
		double MechanicalWBase { get; } // Watt
		double FuelPerHBase { get; } // LiterPerHour

		double ElectricalWAdjusted { get; } // Watt
		double MechanicalWBaseAdjusted { get; } // Watt
		double FuelPerHBaseAdjusted { get; } // LiterPerHour

		void Clone(ISSMTOOL from);

		bool Load(string filePath);

		bool Save(string filePath);

		bool IsEqualTo(ISSMTOOL source);

		/// <summary>
		/// This alters the waste heat and returns an adjusted fueling value
		/// </summary>
		/// <param name="AverageUseableEngineWasteHeatKW"></param>
		/// <returns></returns>
		/// <remarks></remarks>
		double FuelPerHBaseAsjusted(double AverageUseableEngineWasteHeatKW);

		event MessageEventHandler Message;

	}

	public delegate void MessageEventHandler(ref object sender, string message, AdvancedAuxiliaryMessageType messageType);
}